recipes/netcdf/all/conanfile.py

```diff
@@ -0,0 +1,115 @@
+import os
+import glob
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+
+
+class NetcdfConan(ConanFile):
+    name = "netcdf"
+    description = ("NetCDF is a set of software libraries and "
+                   "self-describing, machine-independent data "
+                   "formats that support the creation, access, "
+                   "and sharing of array-oriented scientific data.")
+    license = "BSD-3-Clause"
+    homepage = "https://www.unidata.ucar.edu/software/netcdf/"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("array", "dataset", "scientific")
+
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_netcdf4": [True, False],
+        "with_dap": [True, False],
+        "with_utilities": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_netcdf4": True,
+        "with_dap": True,
+        "with_utilities": True,
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        if self.options.with_netcdf4:
+            self.requires("hdf5/[>=1.8.9]")
+        if self.options.with_dap:
+            self.requires("libcurl/[>=7.18.0]")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "netcdf-c-{}".format(self.version)
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def build(self):
+        args = [
+            "--prefix={}".format(self.package_folder),
+        ]
+
+        if self.options.shared:
+            args.append("--enable-shared")
+            args.append("--disable-static")
+        else:
+            args.append("--disable-shared")
+            args.append("--enable-static")
+
+        if self.options.with_netcdf4:
+            args.append("--enable-netcdf4")
+        else:
+            args.append("--disable-netcdf4")
+
+        if self.options.with_dap:
+            args.append("--enable-dap")
+        else:
+            args.append("--disable-netcdf4")
+
+        if self.options.with_utilities:
+            args.append("--enable-utilities")
+        else:
+            args.append("--disable-utilities")
+
+        if self.settings.compiler != 'Visual Studio' and self.options.fPIC:
+            args.append('--with-pic')
+
+        env_build = AutoToolsBuildEnvironment(self)
```

Comment:
you have to cache the autotools object, when reusing it.

Please, use https://github.com/conan-io/conan-center-index/blob/7fea3121494fbbc312894bae978c12d015abcad4/recipes/libiberty/all/conanfile.py#L46 as reference.

Reply:
Should be cached now.